dali/operators/numba_function/numba_func.cu

```diff
@@ -0,0 +1,222 @@
+    auto& in = ws.Input<GPUBackend>(in_id);
+    in_shapes_[in_id] = in.shape();
+    DALI_ENFORCE(in_shapes_[in_id].sample_dim() == ins_ndim_[in_id], make_string(
+      "Number of dimensions passed in `ins_ndim` at index ", in_id,
+      " doesn't match the number of dimensions of the input data: ",
+      in_shapes_[in_id].sample_dim(), " != ", ins_ndim_[in_id]));
+    DALI_ENFORCE(in.type() == in_types_[in_id], make_string(
+      "Data type passed in `in_types` at index ", in_id, " doesn't match type of the input data: ",
+      in.type(), " != ", in_types_[in_id]));
+  }
+
+  for (size_t in_id = 0; in_id < in_types_.size(); in_id++) {
+    vector<ssize_t> sizes = calc_sizes(in_types_[in_id], in_shapes_[in_id][0]);
+    in_sizes_.push_back(sizes);
+    in_memory_ptrs_.push_back({nullptr, nullptr});
+  }
+
+  for (size_t out_id = 0; out_id < out_types_.size(); out_id++) {
+    // For now we assume that inputs and outputs have the same shapes and types
+    vector<ssize_t> sizes = calc_sizes(in_types_[out_id], in_shapes_[out_id][0]);
+    out_sizes_.push_back(sizes);
+    out_memory_ptrs_.push_back({nullptr, nullptr});
+  }
+
+  for (int i = 0; i < noutputs; i++) {
+    const auto &in = ws.Input<GPUBackend>(i);
+    output_desc[i] = {in.shape(), in.type()};
+  }
+  return true;
+}
+
+
+template <>
+void NumbaFuncImpl<GPUBackend>::RunImpl(workspace_t<GPUBackend> &ws) {
+  auto N = ws.Input<GPUBackend>(0).shape().num_samples();
+  int ninputs = ws.NumInput();
+
+  std::vector<uint64_t> out_ptrs;
+  std::vector<uint64_t> in_ptrs;
+  out_ptrs.resize(N * out_types_.size());
+  in_ptrs.resize(N * in_types_.size());
+  for (size_t out_id = 0; out_id < out_types_.size(); out_id++) {
+    auto& out = ws.Output<GPUBackend>(out_id);
+    for (int i = 0; i < N; i++) {
+      out_ptrs[N * out_id + i] = reinterpret_cast<uint64_t>(out.raw_mutable_tensor(i));
+    }
+  }
+  for (size_t in_id = 0; in_id < in_types_.size(); in_id++) {
+    auto& in = ws.Input<GPUBackend>(in_id);
+    for (int i = 0; i < N; i++) {
+      in_ptrs[N * in_id + i] = reinterpret_cast<uint64_t>(in.raw_tensor(i));
+    }
+  }
+
+  for (int i = 0; i < N; i++) {
+    vector<void*> args;
+    for (size_t in_id = 0; in_id < in_types_.size(); in_id++) {
+      vector<void*> args_local = prepare_args(
+        in_memory_ptrs_[in_id],
+        in_sizes_[in_id],
+        &in_ptrs[N * in_id + i]);
+      args.insert(
+        args.end(),
+        make_move_iterator(args_local.begin()),
+        make_move_iterator(args_local.end()));
+    }
+
+    for (size_t out_id = 0; out_id < out_types_.size(); out_id++) {
+      vector<void*> args_local = prepare_args(
+        out_memory_ptrs_[out_id],
+        out_sizes_[out_id],
+        &out_ptrs[N * out_id + i]);
+      args.insert(
+        args.end(),
+        make_move_iterator(args_local.begin()),
+        make_move_iterator(args_local.end()));
+    }
+
+    CUfunction cufunc = (CUfunction)run_fn_;
+    CUresult result = cuLaunchKernel(
```

Comment:
I think you still need to use `batch_processing_` value here and just error when the `batch_processing_==false`

Reply:
Done